lib/records/fhir.rb

```diff
@@ -385,39 +388,31 @@ def self.medications(prescription, fhir_record, patient, encounter)
           r = fhir_record.entry.find { |e| e.resource.is_a?(FHIR::Condition) && reason_code == e.resource.code.coding[0].code }
           reasons << r unless r.nil?
         end
-        med_order = FHIR::MedicationOrder.new('medicationCodeableConcept' => {
-                                                'coding' => [{
-                                                  'code' => med_data[:codes]['RxNorm'][0],
-                                                  'display' => med_data[:description],
-                                                  'system' => 'http://www.nlm.nih.gov/research/umls/rxnorm'
-                                                }]
-                                              },
-                                              'patient' => { 'reference' => patient.fullUrl.to_s },
-                                              'encounter' => { 'reference' => encounter.fullUrl.to_s },
-                                              'dateWritten' => convert_fhir_date_time(prescription['start_time']),
-                                              'reasonReference' => [],
-                                              'eventHistory' => [])
+        med_order = FHIR::MedicationRequest.new('medicationCodeableConcept' => {
+                                                  'coding' => [{
+                                                    'code' => med_data[:codes]['RxNorm'][0],
+                                                    'display' => med_data[:description],
+                                                    'system' => 'http://www.nlm.nih.gov/research/umls/rxnorm'
+                                                  }]
+                                                },
+                                                'stage' => {
+                                                  'coding' => {
+                                                    'code' => 'original-order',
+                                                    'system' => 'http://hl7.org/fhir/request-stage'
+                                                  }
+                                                },
+                                                'patient' => { 'reference' => patient.fullUrl.to_s },
+                                                'context' => { 'reference' => encounter.fullUrl.to_s },
+                                                'dateWritten' => convert_fhir_date_time(prescription['start_time']),
+                                                'reasonReference' => [])
         reasons.each do |r|
           med_order.reasonReference << FHIR::Reference.new('reference' => r.fullUrl.to_s)
         end
-        if prescription['stop']
-          med_order.status = 'stopped'
-
-          event = FHIR::MedicationOrder::EventHistory.new('status' => 'stopped',
-                                                          'dateTime' => convert_fhir_date_time(prescription['stop']))
```

Comment:
For posterity: MedicationRequest does still have an eventHistory but it seems like the Provenance resource requires more detail than we have to provide here (I assume `agents` & `roles` in the context of ending a medication likely refer to the prescribing physician which Synthea isn't modeling yet). Would be nice to have something here but this is fine for now.